andela/activo-web
# #158454264 Fix redirect for protected routes

#### What does this PR do?
Fixe the error that occurs when a non-authenticated user tries to visit a protected route

#### Decsription of tasks to be completed?
- modify mapstatetoprops function to enable checking for null values
- verify if userdetail is null and redirect if null
- fix interface to match props

#### How should this be manually tested?
- Clone repo or Fetch branch locally.
- Start app by running `yarn start:dev`
- Try visiting any protected route without being authenticated. Example of a protected route is `activo-dev.andela.com:3000/dashboard`
- You should be re-directed back to the auth page, with a request to sign in using with your Andela account.

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#158454264](https://www.pivotaltracker.com/story/show/158454264)

#### Screenshots
<img width="1280" alt="screen shot 2018-06-20 at 3 30 57 am" src="https://user-images.githubusercontent.com/24798364/41634171-6d48dfe0-743a-11e8-8810-e21998293f7a.png">
